app/screens/receive-bitcoin-screen/receive-usd.tsx

```diff
@@ -145,6 +146,45 @@ const ReceiveUsd = () => {
   const { lnUpdate } = useSubscriptionUpdates()
   const [showMemoInput, setShowMemoInput] = useState(false)
   const [showAmountInput, setShowAmountInput] = useState(false)
+  const { timeLeft, startCountdownTimer, resetCountdownTimer } = useCountdownTimer()
+  const timeUntilInvoiceExpires = useMemo(() => {
+    if (invoice) {
+      return (
+        getLightningInvoiceExpiryTime(decodeInvoiceString(invoice.paymentRequest)) -
+        Math.round(Date.now() / 1000)
+      )
+    }
+    return NaN
+  }, [invoice])
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener("change", (nextAppState) => {
+      if (appState.current.match(/inactive|background/) && nextAppState === "active") {
+        if (invoice) {
+          resetCountdownTimer(
+            getLightningInvoiceExpiryTime(decodeInvoiceString(invoice.paymentRequest)) -
+            Math.round(Date.now() / 1000),
+            () => setStatus("expired"),
+          )
+        }
+      }
+      appState.current = nextAppState
+    })
+    return () => {
+      subscription.remove()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why not include ```setStatus``` as one of the dependencies?

Reply:
The intention is for this `useEffect` to only run once when the component mounts as it is setting a callback for when the appstate changes and doesn't need to be rerun when any of the dependencies change.

Reply:
I've changed the way it works so now the hook utilises the dependencies.